lib/rules/no-regex-spaces.js

```diff
@@ -27,41 +48,86 @@ module.exports = {
 
-            if (regexResults !== null) {
-                const count = regexResults[1].length;
+            // Skip if there are no consecutive spaces in the source code
+            if (!/ {2}/u.test(rawPattern)) {
+                return;
+            }
 
-                context.report({
-                    node,
-                    message: "Spaces are hard to count. Use {{{count}}}.",
-                    data: { count },
-                    fix(fixer) {
-                        return fixer.replaceTextRange(
-                            [valueStart + regexResults.index, valueStart + regexResults.index + count],
-                            ` {${count}}`
-                        );
-                    }
-                });
+            let ast;
+
+            try {
+                ast = regExpParser.parsePattern(pattern, 0, pattern.length, flags.includes("u"));
+            } catch (e) {
 
-                /*
-                 * TODO: (platinumazure) Fix message to use rule message
-                 * substitution when api.report is fixed in lib/eslint.js.
-                 */
+                // Ignore regular expressions with syntax errors
+                return;
             }
+
+            const spaces = [];
+
+            regexpp.visitRegExpAST(ast, {
+                onCharacterEnter(character) {
+                    if (
+                        character.parent.type === "Alternative" &&
+                        (character.raw === " " || character.raw === "\\ ")
+                    ) {
+                        spaces.push(character);
+                    }
+                }
+            });
+
+            for (let i = 0; i < spaces.length - 1; i++) {
+                let j = i;
+
+                while (
+                    j < spaces.length - 1 &&
+                    spaces[j].end === spaces[j + 1].start &&
+                    spaces[j + 1].raw === " "
+                ) {
+                    j++;
+                }
+
+                if (j > i) {
+                    const count = j - i + 1;
+
+                    context.report({
+                        node: nodeToReport,
+                        message: "Spaces are hard to count. Use {{{count}}}.",
+                        data: { count },
+                        fix(fixer) {
+                            if (pattern !== rawPattern) {
+                                return null;
+                            }
+                            return fixer.replaceTextRange(
+                                [rawPatternStartRange + spaces[i + 1].start, rawPatternStartRange + spaces[j].end],
+                                `{${count}}`
+                            );
+                        }
+                    });
+
+                    // Report only the first occurence of consecutive spaces
+                    return;
+                }
+            }
+
+            /*
+             * TODO: (platinumazure) Fix message to use rule message
```

Comment:
You are right, this TODO can be removed. Thanks!

Reply:
TODO is removed now :)